components/public-api-server/pkg/apiv1/workspace.go

```diff
@@ -317,12 +317,18 @@ func convertWorkspaceInfo(input *protocol.WorkspaceInfo) (*v1.Workspace, error)
 	if err != nil {
 		return nil, err
 	}
+	// Align to https://github.com/gitpod-io/gitpod/blob/0bb6aaad9d4923d5510de865437f22c7f726833f/components/server/src/workspace/workspace-starter.ts#L1375-L1378
+	contextUrl := input.Workspace.ContextURL
+	if input.Workspace.Context != nil && len(input.Workspace.Context.NormalizedContextURL) > 0 {
+		contextUrl = input.Workspace.Context.NormalizedContextURL
+	}
+
 	return &v1.Workspace{
 		WorkspaceId: input.Workspace.ID,
 		OwnerId:     input.Workspace.OwnerID,
 		ProjectId:   "",
 		Context: &v1.WorkspaceContext{
-			ContextUrl: input.Workspace.ContextURL,
+			ContextUrl: contextUrl,
```

Comment:
Does it make sense to mix them? Should not we have a dedicated property for raw and another for normalized?

I'm not sure what is an idea of api, but could be here is raw, but in details is normalized.

Reply:
Addressed @akosyakov